overlay/etc/ssh/sshd_config

```diff
@@ -5,5 +5,13 @@ AllowTcpForwarding      no
 GatewayPorts            no
 PasswordAuthentication  no
 X11Forwarding           no
+PermitRootLogin         no
+LoginGraceTime          30s
+MaxAuthTries            3
+
+Ciphers chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr
+MACs hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256,umac-128@openssh.com
+KexAlgorithms curve25519-sha256@libssh.org,diffie-hellman-group-exchange-sha256
```

Comment:
While I realize that this documentation is a little out of date can you explain why the key-exchange algos you've specified differ from https://infosec.mozilla.org/guidelines/openssh.html#modern-openssh-67 ?

Additionally, with the Principle of Least Astonishment in mind, what is the risk of locking out people from their existing installations if they happen to upgrade to the latest k3OS (once it is released with this change)?

Reply:
If user SSH clients don't support these ciphers, the user has bigger problems.

Key exchange algorithms from: https://stribika.github.io/2015/01/04/secure-secure-shell.html,
which explains why NIST curves are avoided.

Reply:
People might get locked out by the `PermitRootLogin no` directive. However, that would go against "default" configuration as well as best practices, so I suspect that the population this would impact is small.

Reply:
Well, if a user is getting locked out of a k3OS system with `PermitRootLogin=no` then they are definitely "doing it wrong" because the only user should be `rancher`.

Reply:
If we are going to explicitly allow `diffie-hellman-group-exchange-sha256` shouldn't we also be properly setting up `/etc/ssh/moduli`? Additionally, what is considered best practice policy for generating `/etc/ssh/moduli`? Is it good enough to increase the minimum size combined with a one-time generation per host, or maybe it should be every boot?

Reply:
If that should be the only user, then would `AllowUsers rancher` be out of the question?

Reply:
It is a good reasonable default.

Reply:
> If we are going to explicitly allow `diffie-hellman-group-exchange-sha256` shouldn't we also be properly setting up `/etc/ssh/moduli`? Additionally, what is considered best practice policy for generating `/etc/ssh/moduli`? Is it good enough to increase the minimum size combined with a one-time generation per host, or maybe it should be every boot?

@eriner this has me thinking that we might need something a little more robust than default values in the config files, e.g. some extra `k3os.ssh` config

Reply:
I'm not sure the best place to do that off the top of my head, but the script (be it openrc or else) would look something like this:
```
rm /etc/ssh/moduli
ssh-keygen -G /etc/ssh/moduli.candidate -b 4096
ssh-keygen -T /etc/ssh/moduli -f /etc/moduli.candidate
rm /etc/ssh/moduli.candidate
```

Ideally, we would also generate host keys and this happens only once per install, on first boot.
```
cd /etc/ssh
rm ssh_host_*key*
ssh-keygen -t ed25519 -f ssh_host_ed25519_key -N "" < /dev/null
ssh-keygen -t rsa -b 4096 -f ssh_host_rsa_key -N "" < /dev/null
```

Any suggestions as to where to place these are welcome.

Reply:
> @Eriner this has me thinking that we might need something a little more robust than default values in the config files, e.g. some extra k3os.ssh config

@dweomer I don't think any of these options, if the additional items above are resolved, would require tinkering or configuration from users in `k3os.ssh`. What specifically were you thinking?